### PR TITLE
Add otel tesla library

### PR DESCRIPTION
**Pull request by ricardoccpaiva**
## OpenTelemetry Tesla

Tesla middleware that creates OpenTelemetry spans and injects tracing headers into HTTP requests for Tesla clients.
Currently this library is located at https://github.com/ricardoccpaiva/opentelemetry_tesla

### Checklist

**Acknowledgements**

- [x] I certify that I am authorized to assign ownership of this code to the OpenTelemetry project.
- [x] I acknowledge that ownership of the [Hex](https://hex.pm) package must be given/transferred to the [OpenTelemetry Hex Org](https://hex.pm/orgs/opentelemetry)

**Project Type**

- [x] Instrumentation
- [ ] Exporter
- [ ] Propagator
- [ ] Metapackage
- [ ] Example
- [ ] Util

**Prior Art**

- [ ] Is this an existing project? Includes private libraries being open sourced.
  - [x] Yes
  - [ ] No
- [x] Are there any other packages on [Hex](https://hex.pm) providing similar functionality?
  
**Existing Project**

- [x] Moving from an existing public repository? Provide a link in the description.
- [x] All authors who have contributed code contained in this PR have signed the [CNCF CLA](https://identity.linuxfoundation.org/projects/cncf)

**All Projects**

- [x] Label created (maintainers)
- [x] [labeler.yml](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/.github/labeler.yml) updated
- [ ] CI
  - [x] Elixir or Erlang language file is updated
  - [ ] Test strategy should use the matrix strategy with all supported language/OTP combinations
  - [ ] The job is scoped to only run for relevant code changes
- [x] [Codeowners](https://github.com/open-telemetry/opentelemetry-erlang-contrib/blob/main/CODEOWNERS) adds scoped ownership for project maintainers
- [x] License file (Apache2 - copyright to OpenTelemetry Authors)
- [x] All code complies with the [Otel Specification](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification) and Otel Semantic Conventions - [Trace](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions) [Metrics](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/metrics/semantic_conventions)

### CODE REVIEW

**Comment by ricardoccpaiva** [comment]
when should the hex.pm org be transferred? after merging?

**Comment by ricardoccpaiva** [comment]
@tsloughter @bryannaegele can I get a review on this? 🙏

File: instrumentation/opentelemetry_tesla/lib/middleware/opentelemetry_tesla_middleware.ex [hunk 1]
```diff
@@ -0,0 +1,93 @@
+defmodule Tesla.Middleware.OpenTelemetry do
```
Comment (tsloughter) [inline instrumentation/opentelemetry_tesla/lib/middleware/opentelemetry_tesla_middleware.ex:1]: Is using `Tesla.` as the module prefix, "namespace", the pattern used for Tesla middlewares that are from third parties?

**Comment by tsloughter** [comment]
@ricardoccpaiva sorry about that! I think this is good to go. I just had one question.

File: instrumentation/opentelemetry_tesla/lib/middleware/opentelemetry_tesla_middleware.ex (hunk 1 above)
Comment (nirev) [reply instrumentation/opentelemetry_tesla/lib/middleware/opentelemetry_tesla_middleware.ex:1]: Yes, it is

**Review by bryannaegele — APPROVED** [review]

**Comment by bryannaegele** [comment]
@ricardoccpaiva now that this is merged, can you please transfer package ownership to the `opentelemetry`? Thank you!

**Comment by ricardoccpaiva** [comment]
> @ricardoccpaiva now that this is merged, can you please transfer package ownership to the `opentelemetry`? Thank you!

yes! it should be using the [mix hex.owner](https://hexdocs.pm/hex/Mix.Tasks.Hex.Owner.html) command, right?

**Comment by bryannaegele** [comment]
That's it! It would be `mix hex.owner transfer opentelemetry_tesla opentelemetry`

**Comment by ricardoccpaiva** [comment]
> That's it! It would be `mix hex.owner transfer opentelemetry_tesla opentelemetry`

done ✅

**Pull request merged**